salt/modules/cohesity.py

```diff
@@ -0,0 +1,531 @@
+        RunProtectionJobParam,
+    )
+    from cohesity_management_sdk.models.universal_id import UniversalId
+    from cohesity_management_sdk.models.update_protection_jobs_state_request_body import (
+        UpdateProtectionJobsStateRequestBody,
+    )
+    from cohesity_management_sdk.models.vmware_restore_parameters import (
+        VmwareRestoreParameters,
+    )
+
+    HAS_LIBS = True
+except ImportError as err:
+    HAS_LIBS = False
+
+log = logging.getLogger(__name__)
+__virtualname__ = "cohesity"
+
+
+def _get_client():
+    context_key = "{}.cohesity_client".format(__name__)
+    if context_key in __context__:
+        return __context__[context_key]
+    cohesity_config = __opts__.get("cohesity_config", {})
+    cluster_vip = cohesity_config.get("cluster_vip", "")
+    c_username = cohesity_config.get("username", "")
+    c_password = cohesity_config.get("password", "")
+    c_domain = cohesity_config.get("domain", "")
+    cohesity_client = CohesityClient(
+        cluster_vip=cluster_vip,
+        username=c_username,
+        password=c_password,
+        domain=c_domain,
+    )
+    __context__[context_key] = cohesity_client
+    return cohesity_client
+
+
+def __virtual__():
+    if HAS_LIBS:
+        return __virtualname__
+    return False, "Could not import the Cohesity SDK modules"
+
+
+def get_sd_id(name):
+    """
+    Function to fetch storage domain available in the cluster.
+
+    : return: storage domain id.
+    """
+    cohesity_client = _get_client()
+    log.info("Getting sorage domain with name {}".format(name))
+    resp = cohesity_client.view_boxes.get_view_boxes(names=name)
+    if resp:
+        return resp[0].id
+
+
+def get_policy_id(name):
+    """
+    Function to fetch policy available in the cluster.
+
+    : return policy id.
+    """
+    cohesity_client = _get_client()
+    log.info("Getting policy with name {}".format(name))
+    resp = cohesity_client.protection_policies.get_protection_policies(names=name)
+    if resp:
+        return resp[0].id
+
+
+def get_vmware_source_ids(name, vm_list):
+    """
+    Function to virtual machines available in the vcenter.
+
+    : return source ids and vcenter id.
+    """
+    cohesity_client = _get_client()
+    source_id_list = []
+    parent_id = -1
+    log.info("Fetching Vcenter and Vm ids")
+    try:
+        result = cohesity_client.protection_sources.list_protection_sources_root_nodes(
+            environments=env_enum.K_VMWARE
+        )
+        for each_source in result:
+            endpoint = each_source.registration_info.access_info.endpoint
+            v_name = each_source.protection_source.name
+
+            # Check for both endpoint and source name.
+            if name in [endpoint, v_name]:
+                parent_id = each_source.protection_source.id
+        if parent_id == -1:
+            log.error("Vcenter {} not available in the cluster".format(name))
+            exit()
```

Comment:
```suggestion
            log.error("Vcenter %s not available in the cluster", name)
```
Never call `exit()`, return an error or something else you need to, but never `exit`